tests/mock_network/churn.rs

```diff
@@ -675,7 +675,7 @@ impl Expectations {
             })
             .collect();
         let mut section_msgs_received = HashMap::new(); // The count of received section messages.
-        for node in nodes {
+        for node in nodes.iter_mut() {
```

Comment:
why we now requires a `mut` iterator?
looks like there is no change requiring that?

Reply:
[`try_next_ev`](https://github.com/maidsafe/routing/pull/1986/files#diff-99ea4a9327d9bac124927e60fdbc4e98R682) requires `mut`.

Reply:
but this is not a new change. `try_next_ev` was there previously.

Reply:
No, but `nodes` is a mutable reference, so `for node in nodes` would consume it, but I need to use it again later.

Reply:
Ah, Ok, I get it.